src/precice/Participant.hpp

```diff
@@ -22,27 +22,180 @@ struct WhiteboxAccessor;
+ * skinparam conditionEndStyle hline
+ * start
+ * :Participant();
+ * note right: Create a participant
+ * :setMeshVertices();
+ * note right: Define your meshes
+ * rectangle {
+ * note right
+ * //Define mesh connectivity//
+ * ----
+ * Only required for
+ * * projection mappings
+ * * cell mappings
+ * * watch integrals
+ * * scaled-consistent mappings
+ * end note
+ * if ( requiresMeshConnectivityFor()) then (yes)
+ * :setMeshEdges()
+ * setMeshTriangles()
+ * setMeshTetrahedra();
+ * else (no)
+ * endif
+ * }
+ *
+ * rectangle {
+ *
+ * note right
+ * //Provide initial data//
+ * ----
+ * Only required for non-zero
+ * boundary conditions.
+ * end note
+ *
+ * if (requiresInitialData()) then (yes)
+ * :writeData();
+ * else (no)
+ * endif
+ * }
+ *
+ * :initialize();
+ * note right: Initialize coupling
  *
- * To adapt a solver to preCICE, follow the following main structure:
+ * while (isCouplingOnGoing()) is (yes)
+ *
+ * rectangle {
+ * note right
+ * //Implicit coupling//
+ * ----
+ * New time window
+ * Save solver state
+ * end note
+ * if (requiresWritingCheckpoint()) then (yes)
+ * :solver writes checkpoint;
+ * else (no)
+ * endif
+ * }
+ *
+ * :precice_dt = getMaxTimeStepSize()
+ * solver_dt = solverGetAdaptiveDt()
+ * dt = min(precice_dt, solver_dt);
+ * note right: Agree on time step size
+ *
+ * :readData()
+ * solverDoTimeStep(dt)
+ * writeData()
+ * advance(dt);
+ * note right: Compute time step
+ *
+ * rectangle {
+ * note right
+ * //Implicit coupling//
+ * ----
+ * Iteration didn't converge
+ * Restore solver state
+ * ----
+ * Iteration converged
+ * Move solver to next time window
+ * end note
+ * if (requiresReadingCheckpoint()) then (yes)
+ * :solver reads checkpoint;
+ * else (no)
+ * :solver moves in time;
+ * endif
+ * }
+ *
+ * endwhile (no)
+ *
+ * stop
+ * @enduml
  *
- * -# Create an object of Participant with Participant()
- * -# Initialize preCICE with Participant::initialize()
- * -# Advance to the next (time)step with Participant::advance()
- * -# Finalize preCICE with Participant::finalize()
  *
  *  @note
  *  We use solver, simulation code, and participant as synonyms.
  *  The preferred name in the documentation is participant.
  */
 class PRECICE_API Participant {
 public:
-  ///@name Construction and Configuration
-  ///@{
+  /**
+   * @name Construction and Configuration
+   *
+   * The API of preCICE is accessible via the \ref Participant class.
+   * A constructed \ref Participant directly corresponds to a participant in the configuration file.
+   *
+   * Constructors require defining the parallel context of the Participant by providing
+   * index and size of the current process, which are equivalent to rank and size in the MPI terminology.
+   *
+   * If preCICE is compiled with MPI, then there are multiple ways for it to be used:
+   * 1. if the participant is configured to use an intra-comm using sockets, preCICE ignores MPI
+   * 2. if a custom communicator is provided, preCICE uses it
```

Comment:
Point one and two might be contradicting, what happens in the case where I provide a custom MPI comm and configure sockets?

Reply:
I looked this up again and realised that we always initialize MPI if we define a socket intracomm...

Fishy.

Reply:
Fixed